client/v2/autocli/query_test.go

```diff
@@ -143,6 +143,26 @@ func TestEverything(t *testing.T) {
 	assert.DeepEqual(t, conn.lastRequest, conn.lastResponse.(*testpb.EchoResponse).Request, protocmp.Transform())
 }
 
+func TestJsonFlag(t *testing.T) {
+	conn := testExecCommon(t, buildModuleQueryCommand,
+		"echo",
+		"1", "abc", `{"denom":"foo","amount":"1"}`,
+		"--some-messages", `{"bar":"baz"}`,
+		"-u", "27", // shorthand
+	)
+	lastReq := conn.lastRequest.(*testpb.EchoRequest)
+	assert.Equal(t, uint32(27), lastReq.U32)
+
+	conn = testExecCommon(t, buildModuleQueryCommand,
+		"echo",
+		"1", "abc", `{"denom":"foo","amount":"1"}`,
+		"--some-messages", "testdata/some_message.json",
+		"-u", "27", // shorthand
+	)
+	lastReq = conn.lastRequest.(*testpb.EchoRequest)
+	assert.Equal(t, uint32(27), lastReq.U32)
```

Comment:
That test isn't really testing the good parsing of the json, but only tests that `-u` argument is well-read.
We should update the test to reflect what we want to test.